pkgs/tools/networking/pgrok/web.nix

```diff
@@ -0,0 +1,29 @@
+{ buildNpmPackage
+, src
+, version
+}:
+buildNpmPackage {
+    name = "pgrok-web";
+    inherit src version;
+    sourceRoot = "${src.name}/pgrokd/web";
+
+    npmDepsHash = "sha256-f4pDBoG6sTJE3aUknqUvHHpBR9KWo/B4YMrWHkGbvA8=";
+
+    # Upstream doesn't have a lockfile
+    postPatch = ''
+      cp ${./package-lock.json} ./package-lock.json
+      substituteInPlace ./package.json \
+        --replace "../cli/dist" "$out"
```

Comment:
this looks a bit wired :eyes: 
can you not do something like
```
cp $src/package-lock.json package-lock.json
```

Reply:
Upstream only has a pnpm lockfile, no package-lock.json and because of https://github.com/NixOS/nixpkgs/issues/231513#issuecomment-1642375998 i roll my own lockfile instead

Reply:
Oh I see I was a bit stupid and didn't realize that ${} copy's the file from your local folder but of course it does.